src/Hadolint/Rule.hs

```diff
@@ -21,7 +21,30 @@ data DLSeverity
   | DLInfoC
   | DLStyleC
   | DLIgnoreC
-  deriving (Show, Eq, Ord, Generic, NFData)
+  deriving (Show, Read, Eq, Ord, Generic, NFData)
+
+instance Yaml.FromYAML DLSeverity where
+  parseYAML = withSeverity pure
+
+withSeverity :: (DLSeverity -> Yaml.Parser a) -> Yaml.Node Yaml.Pos -> Yaml.Parser a
+withSeverity f v@(Yaml.Scalar _ (Yaml.SStr b)) =
+  case Hadolint.Rule.readSeverity b of
```

Comment:
I find it odd that you are using a fully qualified call for a function in the same module. Is there a reason for this?

Reply:
Initially I wanted to name the function just `read`, which would have required the fully qualified call, but then I had to rename it because the `read` function already exists for the `LabelType`.
I can probably remove the fully qualified part here and also below in `withLabelType`, if I rename `read` to `readLabelType`.

I might do that in the future. There are also some very similar `veryCustomRule`s which whose state tracking could be unified.